### PR TITLE
Change exceptions used to catch failed GCE authentication

### LINKED ISSUE #333: \Apigee\Edge\HttpClient\Plugin\Authentication\GceServiceAccount::isAvailable() throws exception when called on non-GCE context
## Description

The try/catch inside the mentioned method does not work because a different exception is thrown (on Drupal 10.1) that is not a subclass of the currently catched exception.

## Steps to Reproduce
Build and initiate a GceServiceAccount and try to authenticated with that in a non-GCE hosted environment.

## Actual Behavior
Exception is thrown.

## Expected Behavior
FALSE is returned.

## Screenshots

image

## Version Info
3.0.3

### PR DESCRIPTION
**Pull request by mxr576**
Closes #333

### CODE REVIEW

**Comment by mxr576** [comment]
> Alternative considering, but it feels an overkill to me, catch \Psr\Http\Client\ClientExceptionInterface

Maybe this approach combines best from both worlds, isAvailable() should only fail with `NetworkExceptionInterface`, when the project is not hosted on GCE  -- right? However getAccessToken() can also fail when the service account is not configured properly. so the host was reachable and it returned an HTTP >= 400 code.

File: src/HttpClient/Plugin/Authentication/GceServiceAccount.php [hunk 1]
```diff
@@ -61,7 +62,7 @@ public function isAvailable(): bool
             $this->authClient()->get('');
 
             return true;
-        } catch (Exception $e) {
+        } catch (NetworkExceptionInterface $e) {
```
Comment (shishir-intelli) [inline src/HttpClient/Plugin/Authentication/GceServiceAccount.php:65]: @mxr576 ,
With `NetworkExceptionInterface` I am not able to access the `admin/config/apigee-edge/settings` page for credential settings.

Screenshot :
image

Comment (mxr576) [reply src/HttpClient/Plugin/Authentication/GceServiceAccount.php:65]: Thanks for checking, I think I have found the root cause of that: https://github.com/apigee/apigee-edge-drupal/pull/1019

Comment (shishir-intelli) [reply src/HttpClient/Plugin/Authentication/GceServiceAccount.php:65]: @mxr576 Thankyou

**Review by shishir-intelli — APPROVED** [review]
LGTM

**Pull request merged**